### PR TITLE
implementation of map with open street map map can also rotate and zo…

### PR DESCRIPTION
**Pull request by Neyross**
…om with fingers

### CODE REVIEW

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 1]
```diff
@@ -1,142 +1,54 @@
...
-                    iconSize(0.15)
-                    iconAllowOverlap(false)
-                    iconAnchor(IconAnchor.BOTTOM)
-                }
-            }
-        )
-        return view
+import org.osmdroid.tileprovider.tilesource.TileSourceFactory
+import org.osmdroid.views.MapView
+import org.osmdroid.config.Configuration.*
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:12]: 🚫 **[ktlint]** <no-wildcard-imports> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wildcard import (cannot be auto-corrected)

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 2]
```diff
@@ -1,142 +1,54 @@
...
+import org.osmdroid.util.GeoPoint
+import org.osmdroid.views.overlay.gestures.RotationGestureOverlay
+
+class MapFragment : Fragment(R.layout.fragment_map) {
+    private lateinit var map : MapView
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:17]: 🚫 **[ktlint]** <colon-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected spacing before ":"

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 3]
```diff
@@ -1,142 +1,54 @@
...
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        getInstance().load(context,
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:20]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:20]: 🚫 **[ktlint]** <wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing newline after "("

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 4]
```diff
@@ -1,142 +1,54 @@
...
+            context?.let { PreferenceManager.getDefaultSharedPreferences(it) })
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:21]: 🚫 **[ktlint]** <wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing newline before ")"

Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:21]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing newline before ")"

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 5]
```diff
@@ -1,142 +1,54 @@
...
+        val lay = super.onCreateView(inflater, container, savedInstanceState)
+        map = lay?.findViewById(R.id.map)!!
+        map.setTileSource(TileSourceFactory.MAPNIK)
+
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:25]: 🚫 **[ktlint]** <no-trailing-spaces> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Trailing space(s)

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 6]
```diff
@@ -1,142 +1,54 @@
...
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        getInstance().load(context,
+            context?.let { PreferenceManager.getDefaultSharedPreferences(it) })
+        val lay = super.onCreateView(inflater, container, savedInstanceState)
+        map = lay?.findViewById(R.id.map)!!
+        map.setTileSource(TileSourceFactory.MAPNIK)
+        
+        val mapController = map.controller
+        mapController.setZoom(9.5)
+        val startPoint = GeoPoint(48.8583, 2.2944);
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:28]: 🚫 **[ktlint]** <no-semi> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary semicolon

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 7]
```diff
@@ -1,142 +1,54 @@
...
+        mapController.setCenter(startPoint);
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:29]: 🚫 **[ktlint]** <no-semi> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary semicolon

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 8]
```diff
@@ -1,142 +1,54 @@
...
-        super.onLowMemory()
-        mapView?.onLowMemory()
-    }
-
-    override fun onDestroy() {
-        super.onDestroy()
-        mapView?.onDestroy()
+    override fun onResume() {
+        super.onResume()
+        //this will refresh the osmdroid configuration on resuming.
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:39]: 🚫 **[ktlint]** <comment-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing space after //

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 9]
```diff
@@ -1,142 +1,54 @@
...
+        //if you make changes to the configuration, use
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:40]: 🚫 **[ktlint]** <comment-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing space after //

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 10]
```diff
@@ -1,142 +1,54 @@
...
+        //SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:41]: 🚫 **[ktlint]** <comment-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing space after //

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 11]
```diff
@@ -1,142 +1,54 @@
...
+        //Configuration.getInstance().load(this, PreferenceManager.getDefaultSharedPreferences(this));
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:42]: 🚫 **[ktlint]** <comment-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing space after //

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 12]
```diff
@@ -1,142 +1,54 @@
...
+        map.onResume() //needed for compass, my location overlays, v6.0.0 and up
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:43]: 🚫 **[ktlint]** <comment-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing space after //

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 13]
```diff
@@ -1,142 +1,54 @@
...
-            )
-        } else {
-            val locationRequest = LocationRequest()
-            locationRequest.interval = 2000
-            locationRequest.fastestInterval = 1000
-            locationRequest.priority = LocationRequest.PRIORITY_HIGH_ACCURACY
-        }
+    override fun onPause() {
+        super.onPause()
+        //this will refresh the osmdroid configuration on resuming.
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:48]: 🚫 **[ktlint]** <comment-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing space after //

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 14]
```diff
@@ -1,142 +1,54 @@
...
+        //if you make changes to the configuration, use
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:49]: 🚫 **[ktlint]** <comment-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing space after //

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 15]
```diff
@@ -1,142 +1,54 @@
...
+        //SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:50]: 🚫 **[ktlint]** <comment-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing space after //

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 16]
```diff
@@ -1,142 +1,54 @@
...
+        //Configuration.getInstance().save(this, prefs);
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:51]: 🚫 **[ktlint]** <comment-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing space after //

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 17]
```diff
@@ -1,142 +1,54 @@
...
+        map.onPause()  //needed for compass, my location overlays, v6.0.0 and up
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:52]: 🚫 **[ktlint]** <no-multi-spaces> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary long whitespace

Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:52]: 🚫 **[ktlint]** <comment-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing space after //

**Review by ArthurBoucard — APPROVED** [review]

**Comment by ArthurBoucard** [comment]
J'ai approve mais il manque le linter bien sur 🐶

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 18]
```diff
@@ -1,142 +1,60 @@
...
-            locationRequest.fastestInterval = 1000
-            locationRequest.priority = LocationRequest.PRIORITY_HIGH_ACCURACY
-        }
+    override fun onPause() {
+        super.onPause()
+        // this will refresh the osmdroid configuration on resuming.
+        // if you make changes to the configuration, use
+        // SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
+        // Configuration.getInstance().save(this, prefs);
+        map.onPause()  // needed for compass, my location overlays, v6.0.0 and up
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:58]: 🚫 **[ktlint]** <no-multi-spaces> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary long whitespace

**Review by claraxuxu — APPROVED** [review]

**Review by larroyo3 — APPROVED** [review]
Super !

**Pull request merged**